src/main/java/com/fasttime/domain/member/controller/MemberController.java

```diff
@@ -17,56 +19,73 @@
 @RestController
 @RequiredArgsConstructor
 public class MemberController {
-    private final MemberService memberService;
-
 
+    private final MemberService memberService;
 
     private final MemberRepository memberRepository;
 
-    @PostMapping("/users/new-user") // 회원가입
-    public ResponseEntity<String> join(@Valid @RequestBody MemberDto userDto) { // @Valid 어노테이션 추가
+    @PostMapping("/v1/join")
+    public ResponseEntity<String> join(@Valid @RequestBody MemberDto memberDto) {
         try {
-            memberService.save(userDto);
-            return ResponseEntity.ok("Join success"); // 메시지 명시
-        } catch (DuplicateKeyException e) { // 중복된 이메일인 경우
-            return ResponseEntity.badRequest().body("Join failed: Email already exists");
-        } catch (Exception e) { // 그 외의 예외 처리
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("회원가입 실패: " + e.getMessage());
+            if (memberService.isEmailExistsInFcmember(memberDto.getEmail())) {
+                if (memberService.isEmailExistsInMember(memberDto.getEmail())) {
+                    return ResponseEntity.badRequest().body("이미 가입된 회원입니다.");
+                } else if (memberService.checkDuplicateNickname(memberDto.getNickname())) {
+                    return ResponseEntity.badRequest().body("이미 사용 중인 닉네임 입니다.");
+                }
+
+                memberService.save(memberDto);
+                return ResponseEntity.ok("가입 성공!");
+            } else {
+                return ResponseEntity.badRequest().body("FastCampus에 등록된 이메일이 아닙니다.");
+            }
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                .body("회원가입 실패 " + e.getMessage());
         }
-
     }
 
+    @PutMapping("v1/retouch-member") // 회원 정보 수정
+    public ResponseEntity<EditResponse> updateMember(@RequestBody EditRequest editRequest,
+        HttpSession session) {
+        Member member = (Member) session.getAttribute("member");
+        if (member != null) {
+            // 닉네임 중복 여부 검사
+            if (!member.getNickname().equals(editRequest.getNickname()) &&
+                memberService.checkDuplicateNickname(editRequest.getNickname())) {
+                return ResponseEntity.badRequest().body(new EditResponse("중복된 닉네임입니다."));
+            }
+
+            // 이미지 업데이트
+            member.setImage(editRequest.getImage());
+
+            // 업데이트된 Member 엔티티를 데이터베이스에 저장
+            memberRepository.save(member);
+
+            EditResponse memberResponse = new EditResponse(member);
+            return ResponseEntity.ok(memberResponse);
```

Comment:
Input 검증 + 비즈니스 검증 + DB 저장까지 `Controller`의 역할이 너무 커보입니다. 
몇몇부분은 Service Layer로 분리를 하는게 좋을 것 같습니다.
```suggestion
            if (!member.getNickname().equals(editRequest.getNickname()) &&
                memberService.checkDuplicateNickname(editRequest.getNickname())) {
                return ResponseEntity.badRequest().body(new EditResponse("중복된 닉네임입니다."));
            }

           EditResponse memberResponse =  memberService.updateMember(editRequest);

           return ResponseEntity.ok(memberResponse);
```

Reply:
> Input 검증 + 비즈니스 검증 + DB 저장까지 `Controller`의 역할이 너무 커보입니다. 몇몇부분은 Service Layer로 분리를 하는게 좋을 것 같습니다.

처음에 테스트 코드를 컨트롤러에 한 번에 다 때려 넣어서, 테스트코드에 맞추다 보니 controller에 많은 기능이 들어갔습니다... 후에 수정해보도록 노력하겠습니다😓